In [1]:
%pip install import-ipynb

In [2]:
# Importing the needed Libraries

import import_ipynb
from sklearn.metrics import accuracy_score, confusion_matrix, classification_report

## Logistic Regression Predictive model

In [3]:
# Reading the Stored df from another .ipynb file
%store -r reg_model
%store -r reg_df

In [4]:
X_test = reg_df.iloc[:,:-1]
y_test = reg_df.PerformanceRating

In [5]:
y_pred = reg_model.predict(X_test)
print('Accuracy score: ',round(accuracy_score(y_test, y_pred)*100,2),"%")
print(confusion_matrix(y_test, y_pred))
print(classification_report(y_test, y_pred))

accuracy score:  84.88 %
[[240  20  18]
 [ 29 192  29]
 [  6  17 236]]
              precision    recall  f1-score   support

           2       0.87      0.86      0.87       278
           3       0.84      0.77      0.80       250
           4       0.83      0.91      0.87       259

    accuracy                           0.85       787
   macro avg       0.85      0.85      0.85       787
weighted avg       0.85      0.85      0.85       787



## Random Forest Classifier Predictive model

In [6]:
%store -r rfc_model
%store -r rfc_df

In [7]:
X_test = rfc_df.iloc[:,:-1]
y_test = rfc_df.PerformanceRating

In [8]:
y_pred = rfc_model.predict(X_test)
print('accuracy score: ',round(accuracy_score(y_test, y_pred),2)*100,"%")
print(confusion_matrix(y_test, y_pred))
print(classification_report(y_test, y_pred))

accuracy score:  96.0 %
[[269   8   1]
 [  6 236   8]
 [  1  10 248]]
              precision    recall  f1-score   support

           2       0.97      0.97      0.97       278
           3       0.93      0.94      0.94       250
           4       0.96      0.96      0.96       259

    accuracy                           0.96       787
   macro avg       0.96      0.96      0.96       787
weighted avg       0.96      0.96      0.96       787



## Decision Tree Classifier Predictive model

In [9]:
%store -r dtc_model
%store -r dtc_df

In [10]:
X_test = rfc_df.iloc[:,:-1]
y_test = rfc_df.PerformanceRating

In [11]:
y_pred = dtc_model.predict(X_test)
print('accuracy score: ',round(accuracy_score(y_test, y_pred),2)*100,"%")
print(confusion_matrix(y_test, y_pred))
print(classification_report(y_test, y_pred))

accuracy score:  91.0 %
[[269   4   5]
 [ 11 205  34]
 [  2  12 245]]
              precision    recall  f1-score   support

           2       0.95      0.97      0.96       278
           3       0.93      0.82      0.87       250
           4       0.86      0.95      0.90       259

    accuracy                           0.91       787
   macro avg       0.91      0.91      0.91       787
weighted avg       0.92      0.91      0.91       787



# K Nearest Neighbors Classifier Predictive model

In [12]:
%store -r knn_model
%store -r knn_df

In [13]:
X_test = knn_df.iloc[:,:-1]
y_test = knn_df.PerformanceRating

In [14]:
y_pred = knn_model.predict(X_test)
print('accuracy score: ',round(accuracy_score(y_test, y_pred),2)*100,"%")
print(confusion_matrix(y_test, y_pred))
print(classification_report(y_test, y_pred))

accuracy score:  81.0 %
[[158   3   6]
 [ 54  97  32]
 [  1   2 172]]
              precision    recall  f1-score   support

           2       0.74      0.95      0.83       167
           3       0.95      0.53      0.68       183
           4       0.82      0.98      0.89       175

    accuracy                           0.81       525
   macro avg       0.84      0.82      0.80       525
weighted avg       0.84      0.81      0.80       525



# Multinomial Naive Bayes Predictive model

In [15]:
%store -r nb_model_Multi
%store -r nb_Multi_df

In [16]:
X_test = nb_Multi_df.iloc[:,:-1]
y_test = nb_Multi_df.PerformanceRating

In [17]:
print('accuracy score: ',accuracy_score(y_test, y_pred)*100,"%")
print(confusion_matrix(y_test, y_pred))
print(classification_report(y_test, y_pred))
y_pred = nb_model_Multi.predict(X_test)

accuracy score:  34.66666666666667 %
[[79 40 65]
 [73 29 71]
 [61 33 74]]
              precision    recall  f1-score   support

           2       0.37      0.43      0.40       184
           3       0.28      0.17      0.21       173
           4       0.35      0.44      0.39       168

    accuracy                           0.35       525
   macro avg       0.34      0.35      0.33       525
weighted avg       0.34      0.35      0.33       525



# Gaussian Naive Bayes Predictive model

In [18]:
%store -r nb_model_Gauss
%store -r nb_Gauss_df

In [19]:
X_test = nb_Gauss_df.iloc[:,:-1]
y_test = nb_Gauss_df.PerformanceRating

In [20]:
y_pred = nb_model_Gauss.predict(X_test)
print('accuracy score: ',accuracy_score(y_test, y_pred)*100,"%")
print(confusion_matrix(y_test, y_pred))
print(classification_report(y_test, y_pred))

accuracy score:  91.66666666666666 %
[[5 0 0]
 [0 3 0]
 [0 1 3]]
              precision    recall  f1-score   support

           2       1.00      1.00      1.00         5
           3       0.75      1.00      0.86         3
           4       1.00      0.75      0.86         4

    accuracy                           0.92        12
   macro avg       0.92      0.92      0.90        12
weighted avg       0.94      0.92      0.92        12



# Bernoulli Naive Bayes Predictive model

In [21]:
%store -r nb_model_Bern
%store -r nb_Bern_df

In [22]:
X_test = nb_Bern_df.iloc[:,:-1]
y_test = nb_Bern_df.PerformanceRating

In [23]:
y_pred = nb_model_Bern.predict(X_test)
print('accuracy score: ',accuracy_score(y_test, y_pred)*100,"%")
print(confusion_matrix(y_test, y_pred))
print(classification_report(y_test, y_pred))

accuracy score:  80.0 %
[[8 0 0]
 [1 3 1]
 [1 0 1]]
              precision    recall  f1-score   support

           2       0.80      1.00      0.89         8
           3       1.00      0.60      0.75         5
           4       0.50      0.50      0.50         2

    accuracy                           0.80        15
   macro avg       0.77      0.70      0.71        15
weighted avg       0.83      0.80      0.79        15



Model Training and Evaluation We have tried with different models such as:

1) Logistic Regression (Accuracy – 85.01%)

2) **Random Forest Classifier** (Accuracy - **95%**)

3) Decision Tree Classifier (Accuracy - 92%)

4) KNN (Accuracy - 82%)

5) MultinomialNB (Accuracy - 60%)

6) GaussianNB (Accuracy - 91.6%)

7) BernoulliNB (Accuracy - 86%)